### PR TITLE
Drop 3.7 and add 3.11 Python support.

### PR DESCRIPTION
**Pull request by YevheniiSemendiak**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -65,19 +65,18 @@ jobs:
...
-        - python-version: '3.8'
-          os: windows
         - python-version: '3.9'
           os: macos
         - python-version: '3.9'
           os: windows
+        - python-version: '3.10'
+          os: macos
+        - python-version: '3.10'
+          os: ubuntu
```
Comment (dalazx) [inline .github/workflows/ci.yaml:79]: why ubuntu here?

File: CHANGELOG.D/2903.feature [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Added support of cluster energy schedules.
+`neuro config show --energy` will display awailable energy schedule timeslots.
```
Comment (dalazx) [inline CHANGELOG.D/2903.feature:2]:
```suggestion
`neuro config show --energy` will display awailable energy schedule periods.
```

Comment (YevheniiSemendiak) [reply CHANGELOG.D/2903.feature]: done in https://github.com/neuro-inc/neuro-cli/pull/2907, thx!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (YevheniiSemendiak) [reply .github/workflows/ci.yaml:79]: The logic is as follows:
- test for all OSes on edge versions (latest and earliest available).
- test for a single OS for inner versions. Here I just picked ubuntu for "3.9" and Windows for "3.10" (mostly randomly, to be honest).

**Review by dalazx — APPROVED** [review]
👍

**Pull request merged**